src/QirRuntime/test/QIR-static/qir-test-ouput.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include <iostream>
+#include <sstream>
+
+#include "catch.hpp"
+
+#include "qirTypes.hpp"
+#include "quantum__qis_internal.hpp"
+
+extern "C" void Microsoft__Quantum__Testing__QIR__Out__MessageTest__body(QirString*);        // NOLINT
+
+
+// https://stackoverflow.com/a/5419388/6362941
+// https://github.com/microsoft/qsharp-runtime/pull/511#discussion_r574170031
+// https://github.com/microsoft/qsharp-runtime/pull/511#discussion_r574194191
+struct QOstreamRedirector
+{
+    QOstreamRedirector(std::ostream & newOstream)
+        : old(Quantum::Qis::Internal::SetOutputStream(newOstream))
+    {}
+
+    ~QOstreamRedirector()
+    {
+        Quantum::Qis::Internal::SetOutputStream(old);
+    }
+
+private:
+    std::ostream& old;
+};
+
+
+TEST_CASE("QIR: Out.Message", "[qir.Out][qir.Out.Message]")
+{
+    const std::string       testStr1 = "Test String 1";
+    const std::string       testStr2 = "Test String 2";
+
+    std::ostringstream      outStrStream;
+
+    {
+        QOstreamRedirector qOStreamRedirector(outStrStream);    // Redirect the output from std::cout to outStrStream.
+
+        // Log something (to the redirected output):
+        QirString qstr{std::string(testStr1)};
+        Microsoft__Quantum__Testing__QIR__Out__MessageTest__body(&qstr);
+        qstr.str = testStr2;
+        Microsoft__Quantum__Testing__QIR__Out__MessageTest__body(&qstr);
+
+    } // Recover the output stream.
+
+    REQUIRE(outStrStream.str() == (testStr1 + "\n" + testStr2 + "\n"));
```

Comment:
Is your intent here to test for multiple instances of output to be separated by new line? Which is valid, but you might also consider using SECTIONs to run a few tests with same setup.

Reply:
The intent was to test that 
* the output actually takes place (not ignored),
* the output corresponds to the expected one (is not random or unrelated),
* the multiple outputs are separated with a new line char (not merged/concatenated into a single line).

If with SECTION you mean `TEST_CASE( .. ) { .. }` then I'm not sure about the value it adds if we run SECTIONs with the output redirected. Can you please elaborate?
(I was assuming that I need to test that a single Q# call to `Message()` and a few consecutive Q# calls to `Message()` work as expected)